### PR TITLE
Johnshew dev

### PR DESCRIPTION
**Pull request by johnshew**
Fixed unit to remove async on describe.
Removed integration test.
Should run and pass unit tests in Azure Dev Ops.
Outdated npm modules to current.
Fixed extra call on users loading.
Fixed task update bug.
Better expiration date handling.
More and better error handling throughout - especially for login edge cases.

### CODE REVIEW

File: src/officeGraph.ts [hunk 1]
```diff
@@ -19,11 +19,11 @@ export class OfficeGraph {
                 let data = await response.json();
                 return resolve(data);
             }
-            return reject(new Error(`GET for ${url} failed with ${response.status} ${response.statusText}`));
+            return reject(new Error(`GET for ${url} failed with ${response.status} ${response.statusText} and token ${accessToken.substring(0, 5)}`));
```
Comment (Jatrott) [inline src/officeGraph.ts]: would a fixed separator between status and statusText help with reading?

Comment (Jatrott) [reply src/officeGraph.ts]: same lower down also.

Comment (johnshew) [reply src/officeGraph.ts]: status is an http status number like 404 and status text is something like "not found".  

GET for http://... failed with 400 not found...

**Comment by johnshew** [comment]
Fixes #14

**Pull request merged**